lib/katello/tasks/upgrades/3.8/clear_checksum_type.rake

```diff
@@ -5,24 +5,22 @@ namespace :katello do
       task :clear_checksum_type => %w(environment) do
         User.current = User.anonymous_admin
 
-        Katello::Repository.yum_type.find_each do |repo|
-          repo.transaction do
+        Katello::RootRepository.yum_type.find_each do |root_repo|
+          root_repo.transaction do
             begin
-              if repo.on_demand? && repo.url.present?
-                repo.update_attribute(:checksum_type, nil)
+              if root_repo.on_demand? && root_repo.url.present?
+                root_repo.update_attribute(:checksum_type, nil)
```

Comment:
I think we also need to do this for each `repo.saved_checksum_type`?

Reply:
I'm also not sure if we need to clear out the root repo's checksum type. @jlsherrill would know

Reply:
/me digs around in his memory.  Yes we need to set the saved_checksum_type to nil on the repositories.  Do we need to update the root_repo's checksum type....  I believe so.  

Good catch @akofink !

Reply:
Thanks, handled below when going through the individual repos